pkg/fedresolver/fedschema.go

```diff
@@ -0,0 +1,175 @@
+package fedresolver
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// Define the httpClient interface
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// clientWrapper is a generic wrapper around http.Client or any httpClient implementation
+type clientWrapper struct {
+	client httpClient
+}
+
+func (c *clientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// Your SearchSchema struct with an interface for the HTTP client
+type SearchSchema struct {
+	routeToken []map[string]string
+	client     httpClient
```

Comment:
**For discussion:**
I believe we'll need a client per target ACM Hub.
- There's an initial cost creating the connection, so we want to avoid repeating it.
- I don't think that a single client can connect to multiple endpoints at the same time, so it's likely this will serialize the requests.